octodns/provider/ns1.py

```diff
@@ -234,15 +267,16 @@ class Ns1Provider(BaseProvider):
                'TK', 'TO', 'TV', 'WF', 'WS'},
     }
 
-    def __init__(self, id, api_key, retry_count=4, monitor_regions=None, *args,
-                 **kwargs):
+    def __init__(self, id, api_key, retry_count=4, monitor_regions=None,
+                 parallelism=None, *args, **kwargs):
```

Comment:
We could potentially look at passing the `manager` object down into the providers when they're created to give them an option to inspect things like `max_workers`. Anything following the `*args, **kwargs` pattern here would just ignore it if it didn't want it. We can explore that another time, this is probably fine for now and we'd probably want it to allow things to be explicitly set for cases where you might be running multiple sync's in parallel etc.

Reply:
absolutely agreed